### ISSUE TITLE
Save config in localstorage

### ISSUE THREAD

**Issue by thecyberd3m0n** [inolen/quakejs#28]
I think it's important to remember player's changes into it. It's highly annoying when I must redefine controls ,player name and graphics each game...

**Comment by thecyberd3m0n** [comment]
I see in logs it has "virtual filesystem" undernear. How we can access it outside ioq3?

**Comment by inolen** [comment]
The project mounts an IDBFS instance at startup (which ideally should persist config changes):
https://github.com/inolen/ioq3/blob/master/code/sys/sys_browser.js#L295

**Comment by thecyberd3m0n** [comment]
I used build from quakejs site with own server. Every player connected as UnnamedPlayer, because there was no way to store his configuration. Build from github just didn't work. I'll do some additional analysis later, and give you some logs.
Anyway, how can we access the filesystem deployed by that virtual machine outside of it? I think that would be useful for adding features to Quake ;)

**Comment by ENT108** [comment]
You can edit .pk3 and add your config there and then run ```\exec <your_config>```. That's one thing. I am working on saving feature... but this may take time, IDBFS is new to me.

**Comment by Andacious** [comment]
Has anyone found a way around this? I host a server to play with coworkers and their number 1 gripe is that they lose their keybindings every time. Has anyone even made some partial progress that others can build upon?

**Comment by ENT108** [comment]
Nope. As a workaround, I put player's customized configs to the server. So anyone can load own config upon join. The downside is maintanance of the configs.

**Comment by twitwi** [comment]
if anyone bumps on this issue, by default q3 execs a file call `default.cfg`, and you can both save the config to a file (in the local storage) and execute it later.
In practice:
- join a game,
- make your configuration,
- bring up the console with the ² keys (below escape)
- save your config by typing `writeconfig default`
- refresh to check it worked

you can also have alternate config, e.g., `writeconfig abby` then on refresh, `exec abby`

**Comment by twitwi** [comment]
A trick to inject an existing config: open the browser console and type for instance:

```js
FS.writeFile('base/baseq3/abby.cfg', `
bind k "kill"
name AbBy
`)
```
